### PR TITLE
passkey name

### LINKED ISSUE #23: Security: validate and sanitize PasskeyEnrollSubmission.name (finding 3.2)
## Background

Security review finding **3.2** — the user-supplied `name` field in `PasskeyEnrollSubmission` is stored in `passkeys` without any length or character validation.

## Risk

- **Oversized input:** an arbitrarily long name can waste storage or trigger column-length errors at the database level (ugly 500 instead of a clean 400).
- **Malformed data in the UI:** if the name is rendered in the central-ui without escaping, a crafted name can be a stored XSS vector.

## Proposed fix

In `ConversationService.finishPasskeyEnroll`, before persisting the passkey, validate `submission.name`:

```scala
val name = submission.name.trim
if name.isEmpty || name.length > 64 || !name.matches("[\\p{L}\\p{N} _\\-'.]+")
then ZIO.fail(ConversationError.IllegalState)
else // proceed
```

Adjust the allowed character set and max length to match product requirements.

## Scope / tasks

- [ ] Define the allowed character set and maximum length (suggest: ≤ 64 chars, printable Unicode letters/digits/common punctuation)
- [ ] Add the validation guard in `finishPasskeyEnroll`
- [ ] Return `ConversationResult.IllegalState` (or a dedicated validation error) on failure
- [ ] Add tests: empty name, name > max length, disallowed characters, valid name

### PR DESCRIPTION
**Pull request by goshacodes**
fix #23

### CODE REVIEW

File: auth/src/main/scala/versola/oauth/conversation/ConversationController.scala [hunk 1]
```diff
@@ -186,10 +186,15 @@ object ConversationController extends Controller:
...
 
+  private val PasskeyNameRegex = "^[\\p{L}\\p{N} ()-]+$"
+
   given FormDecoder[PasskeyEnrollSubmission] = (form: Form) =>
     for
       response <- FormDecoder.single[String](form, "response", Right(_))
-      name     <- FormDecoder.optional[String](form, "name", Right(_))
+      name     <- FormDecoder.single[String](form, "name", n =>
+        if n.matches(PasskeyNameRegex) then Right(n)
+        else Left("Invalid passkey name: only letters, digits, and hyphens are allowed"),
```
[the remark this answers is not in the document]
Comment (goshacodes) [reply auth/src/main/scala/versola/oauth/conversation/ConversationController.scala]: fixed

File: auth/src/main/scala/versola/oauth/conversation/ConversationController.scala [hunk 2]
```diff
@@ -186,10 +186,15 @@ object ConversationController extends Controller:
...
       .map(PasskeyAssertionSubmission(_))
 
+  private val PasskeyNameRegex = "^[\\p{L}\\p{N} ()-]+$"
+
   given FormDecoder[PasskeyEnrollSubmission] = (form: Form) =>
     for
       response <- FormDecoder.single[String](form, "response", Right(_))
-      name     <- FormDecoder.optional[String](form, "name", Right(_))
+      name     <- FormDecoder.single[String](form, "name", n =>
+        if n.matches(PasskeyNameRegex) then Right(n)
```
[the remark this answers is not in the document]
Comment (goshacodes) [reply auth/src/main/scala/versola/oauth/conversation/ConversationController.scala]: fixed

**Comment by goshacodes** [comment]
augment review

File: central-ui/forms/passkey-enroll/passkey-enroll.tsx [hunk 3]
```diff
@@ -70,14 +70,24 @@ function PasskeyEnrollForm(props: { config: FormConfig }) {
...
+  const [nameError, setNameError] = createSignal(false);
+
+  const NAME_REGEX = /^[\p{L}\p{N} ()-]+$/u;
 
   const handleEnroll = async () => {
     if (busy()) return;
+    const currentName = name();
+    if (!NAME_REGEX.test(currentName) || currentName !== currentName.trim()) {
+      setNameError(true);
+      return;
```
[the remark this answers is not in the document]
Comment (goshacodes) [reply central-ui/forms/passkey-enroll/passkey-enroll.tsx:84]: fixed
